doc/dev/adapters.md

```diff
@@ -1,15 +1,699 @@
+or custom infrastructure backends.
+
+![mixer architecture](./img/mixer%20architecture.svg)
+
+## Templates, Adapters and Operator Configuration
+
+Mixer structures its incoming attribute data into a more useful form for backends (for example, metrics data) using
+templates. Templates describe the form of data dispatched to adapters when processing a request and the interface that
+the adapter must implement to receive this data. Mixer provides a range of default templates suitable for most use
+cases, which are built into the Mixer binary - you can see a complete list in
+[built-in templates](#heading=h.dzc1vsgx8s4n). We strongly recommend that, when implementing adapters, you use Mixer's
+default templates. Though if they are not suitable for your particular needs you can also create your own templates
+along with adapters to consume the relevant data.
+
+Adapters are implemented with respect to a particular set of templates. Each adapter can register itself with the Mixer
+to process data collected by Mixer for its chosen templates. Istio itself includes some [built-in adapters](https://github.com/istio/mixer/tree/master/adapter)
+by default, but users may need to implement their own to let Mixer send data to their chosen backend.
+
+Operator configuration determines the values of the instances dispatched to an adapter for a template.  Refer to
+[Mixer operator configuration](https://istio.io/docs/concepts/policy-and-control/mixer-config.html#handlers) for details.
+
+The roles of the *template* author, adapter author, and the operator can be summarized as:
+
+* The *template* author defines a *template*. A *template* describes the data Mixer dispatches to adapters, and the
+  interface that the adapter must implement to process that data.. The supported set of *templates* within Mixer
+  determine the various types of data an operator can configure Mixer to create and dispatch to the adapters.
+
+* The adapter author selects the *templates* they want to support based on the functionality the adapter must provide.
+  The adapter author's role is to implement the required set of template-specific interfaces to process the data
+  dispatched by Mixer at runtime.
+
+* The operator defines instances, handlers, and actions. The operator decides what data should be collected
+  ([instances](https://istio.io/docs/concepts/policy-and-control/mixer-config.html#instances)), where it can be sent
+  ([handlers](https://istio.io/docs/concepts/policy-and-control/mixer-config.html#handlers)), and when to send it
+  ([rules](https://istio.io/docs/concepts/policy-and-control/mixer-config.html#rules)).
+
+![operator, adapter and template devs](./img/operator%20template%20adapter%20dev.svg)
+
+Now let's look more closely at how adapters work.
+
+# Template overview
+
+To understand how an adapter instance receives and processes a *template's* specific instances, this section first
```

Comment:
"how an adapter receives", we don't want to talk about "adapter instances" since that's confusing.